core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStore.java

```diff
@@ -179,25 +146,31 @@ private static PageId getPageIdFromKey(byte[] key) {
   }
 
   @Override
-  public long pages() {
-    return mSize.get();
-  }
-
-  @Override
-  public long bytes() {
-    return mBytes.get();
-  }
-
-  @Override
-  public Collection<PageInfo> getPages() {
-    try (RocksIterator iter = mDb.newIterator()) {
-      return Streams.stream(new PageIterator(iter)).collect(Collectors.toList());
+  public boolean restore(Predicate<PageInfo> initFunc) {
+    try {
+      byte[] confData = mDb.get(CONF_KEY);
+      Cache.PRocksPageStoreOptions pOptions = mOptions.toProto();
+      if (confData != null) {
+        Cache.PRocksPageStoreOptions persistedOptions =
+            Cache.PRocksPageStoreOptions.parseFrom(confData);
+        if (!persistedOptions.equals(pOptions)) {
+          mDb.close();
+          return false;
+        }
+      }
+      mDb.put(CONF_KEY, pOptions.toByteArray());
+      try (RocksIterator iter = mDb.newIterator()) {
+        return Streams.stream(new PageIterator(iter)).allMatch(initFunc::test);
+      }
+    } catch (RocksDBException | IOException e) {
+      LOG.error("Failed to restore RocksPageStore:", e);
```

Comment:
We should close the db whenever restore returns false.

Reply:
done